core/src/utils/periodic_task_test.cpp

```diff
@@ -418,4 +418,39 @@ UTEST_F(PeriodicTaskLog, ErrorLog) {
     task.Stop();
 }
 
+UTEST(PeriodicTask, EnableDisable) {
```

Comment:
Add a test that after `enabled=false` + `enabled=on` the task does not always start an iteration instantly, but awaits the configured interval since the last time it was run